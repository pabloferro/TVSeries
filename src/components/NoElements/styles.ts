import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    padding: 40,
  },
  message: {
    textAlign: 'center',
  },
});
