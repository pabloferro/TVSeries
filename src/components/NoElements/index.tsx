import React, {ComponentProps} from 'react';
import {View} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import {brandWarning} from '../../constants/colors';
import CustomText from '../CustomText';
import {TextVariants} from '../CustomText/TextVariants';

import styles from './styles';

interface Props {
  message: string;
  iconName: ComponentProps<typeof MaterialCommunityIcons>['name'];
}
export default function NoElements({message, iconName}: Props) {
  return (
    <View style={styles.container}>
      <MaterialCommunityIcons name={iconName} color={brandWarning} size={60} />
      <CustomText variant={TextVariants.h2} style={styles.message}>
        {message}
      </CustomText>
    </View>
  );
}
