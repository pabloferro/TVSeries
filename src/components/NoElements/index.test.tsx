import React from 'react';
import NoElements from '.';
import {render} from '@testing-library/react-native';

test('renders correctly', () => {
  const {toJSON} = render(
    <NoElements message="No elements" iconName="search" />,
  );

  expect(toJSON()).toMatchSnapshot();
});
