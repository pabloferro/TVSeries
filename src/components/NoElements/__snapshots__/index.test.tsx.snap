// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  MaterialCommunityIcons {"name":"search","color":"#E8D2AE","size":60}
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 18,
          "fontWeight": "500",
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    No elements
  </Text>
</View>
`;
